src/main/java/org/springframework/integration/kafka/support/ProducerListener.java

```diff
@@ -1,55 +0,0 @@
-/*
- * Copyright 2015 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.springframework.integration.kafka.support;
-
-import org.apache.kafka.clients.producer.RecordMetadata;
-
-/**
- * Listener for handling outbound Kafka messages. Exactly one of its methods will be invoked, depending on whether
- * the write has been acknowledged or not.
- *
- * Its main goal is to provide a stateless singleton delegate for {@link org.apache.kafka.clients.producer.Callback}s,
- * which, in all but the most trivial cases, requires creating a separate instance per message.
- *
- * @see org.apache.kafka.clients.producer.Callback
- * @author Marius Bogoevici
- * @since 1.3
- */
-public interface ProducerListener {
```

Comment:
Is there any intent to re-introduce this later? Or is it just going away? I still think it is a good feature to have, especially in the context that the Callback needs to be registered with each call (and typically requires to be created per each send operation (since it doesn't retain information about the outgoing data).